openqaoa/workflows/optimizer.py

```diff
@@ -682,7 +682,10 @@ def optimize(self, verbose=False):
         problem = self.problem  
         n_cutoff = self.rqaoa_parameters.n_cutoff
         n_qubits = problem.n
-        counter = self.rqaoa_parameters.counter 
+        counter = self.rqaoa_parameters.counter
+
+        # copy the original qaoa object
+        q = copy.deepcopy(self._q)
```

Comment:
perhaps it makes sense to use a more descriptive name for this? `q` does not tell me that this is the QAOA with the original problem.

Reply:
but this q is going to change at each RQAOA step

Reply:
Exactly